README.md

```diff
@@ -69,17 +69,64 @@ application restarts.
 
 ### Use correct application data directories
 
-Modify Qt application main function by setting the organization and application name
-to match the declaration added to the desktop file.
+Applications that are not based on the Sailfish App library
+([libsailfishapp](https://github.com/sailfishos/libsailfishapp))
+must ensure that correct organization and application name is set.
 
-    QScopedPointer<QGuiApplication> app(Sailfish::createApplication(argc, argv));
     ...
+    QGuiApplication *app = ...;
     app->setOrganizationName(QStringLiteral("org.foobar"));
     app->setApplicationName(QStringLiteral("MyApp"));
     ...
 
-When Sailfish App library (libsailfishapp) is used these values are set automatically to the values
-set in desktop file.
+When Sailfish App library is used, these values are set automatically to the values
+set in application's Desktop Entry file.
+
+Sandboxed applications cannot use `QSettings` constructed with the default arguments - it would use
+an inaccessible path outside `QStandardPaths::AppDataLocation`. A suitable settings file path needs
+to be determined explicitly.
+
+    const QString settingsPath =
+        QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)
+        + "/" + QCoreApplication::applicationName() + ".conf";
+    QSettings settings(settingsPath, QSettings::NativeFormat);
+
+Unless you are creating a new application, it is also necessary to take care about migrating
+application data and configuration from the old location outside the sandbox.
+
+An example of how `QSettings`-based configuration may be migrated:
+
+    if (!settings.contains("migrated")) {
+        const QString oldSettingsPath =
+            QStandardPaths::writableLocation(QStandardPaths::ConfigLocation)
+            + "/" + QCoreApplication::applicationName()
+            + "/" + QCoreApplication::applicationName() + ".conf";
+        QSettings oldSettings(oldSettingsPath, QSettings::NativeFormat);
+
```

Comment:
Yup, I noticed your suggestion on Forum and actually I initially I liked it more than Ville's approach, but after all chose to go with Ville's as it feels a bit more versatile.

Reply:
Versatile why hat?

Reply:
You can update the values as you copy them (some may contain paths), you can deal with possible failure (who knows when OOM killer steps in etc.) because successful migration is clearly determined by storing some config value at the end (with your approach you would have to copy files, update config then remove original files or so), etc.